src/benchmarks/micro/libraries/Microsoft.Extensions.DependencyInjection/ActivatorUtilitiesBenchmark.cs

```diff
@@ -5,70 +5,372 @@
+            collection.AddTransient<TypeWith3ParametersToBeActivated>();
+            collection.AddTransient<TypeWith4ParametersToBeActivated>();
+            collection.AddTransient<TypeWith5ParametersToBeActivated>();
+
             collection.AddSingleton<DependencyA>();
             collection.AddSingleton<DependencyB>();
             collection.AddSingleton<DependencyC>();
-            collection.AddTransient<TypeToBeActivated>();
+            collection.AddSingleton<DependencyD>();
+            collection.AddSingleton<DependencyE>();
+
+            _types2 = new Type[] { typeof(DependencyA), typeof(DependencyB) };
+            _types2_OutOfOrder = new Type[] { typeof(DependencyB), typeof(DependencyC) };
+            _types3 = new Type[] { typeof(DependencyA), typeof(DependencyB), typeof(DependencyC) };
 
+            _factory2 = ActivatorUtilities.CreateFactory(typeof(TypeWith3ParametersToBeActivated), _types2);
+            _factory2_OutOfOrder = ActivatorUtilities.CreateFactory(typeof(TypeWith3ParametersToBeActivated), _types2_OutOfOrder);
+            _factory3 = ActivatorUtilities.CreateFactory(typeof(TypeWith3ParametersToBeActivated), _types3);
+
+            _factoryArguments0 = Array.Empty<object>();
+            _factoryArguments1 = new object[] { new DependencyA() };
+            _factoryArguments2 = new object[] { new DependencyA(), new DependencyB() };
+            _factoryArguments2_OutOfOrder = new object[] { new DependencyB(), new DependencyC() };
+            _factoryArguments3 = new object[] { new DependencyA(), new DependencyB(), new DependencyC() };
+            _factoryArguments4 = new object[] { new DependencyA(), new DependencyB(), new DependencyC(), new DependencyD() };
+            _factoryArguments5 = new object[] { new DependencyA(), new DependencyB(), new DependencyC(), new DependencyD(), new DependencyE() };
             _serviceProvider = collection.BuildServiceProvider();
-            _factory = ActivatorUtilities.CreateFactory(typeof(TypeToBeActivated), new Type[] { typeof(DependencyB), typeof(DependencyC) });
-            _factoryArguments = new object[] { new DependencyB(), new DependencyC() };
         }
 
         [GlobalCleanup]
         public void Cleanup() => _serviceProvider.Dispose();
 
         [Benchmark]
         [BenchmarkCategory(Categories.NoAOT)]
-        public TypeToBeActivated ServiceProvider() => _serviceProvider.GetService<TypeToBeActivated>();
+        public TypeWith0ParametersToBeActivated GetService_0Injected()
+        {
+            return _serviceProvider.GetService<TypeWith0ParametersToBeActivated>();
+        }
+
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public TypeWith1ParameterToBeActivated GetService_1Injected()
+        {
+            TypeWith1ParameterToBeActivated obj = _serviceProvider.GetService<TypeWith1ParameterToBeActivated>();
+            if (obj._a is null)
+            {
+                throw new Exception("Not expected.");
```

Comment:
Could we use assertions instead? Assert.NotNull(obj._a);

Reply:
I removed these checks; Assert wouldn't work since benchmarks aren't run in debug build.

Reply:
Ok, such checks probably already covered in unit tests